src/likelihoods/gaussian.jl

```diff
@@ -20,3 +20,23 @@ GaussianLikelihood() = GaussianLikelihood(1e-6)
 (l::GaussianLikelihood)(f::Real) = Normal(f, sqrt(l.σ²))
 
 (l::GaussianLikelihood)(fs::AbstractVector{<:Real}) = MvNormal(fs, sqrt(l.σ²))
+
+"""
+    HeteroscedasticGaussianLikelihood(σ²)
+
+Heteroscedastic Gaussian likelihood. This is to be used if we assume that the 
+uncertainity associated with the data follows a Gaussian distribution where the 
+variance varies according to the input.
+
+```math
+    p(y|[f, σ]) = Normal(y | f, σ)
+```
+On calling, this would return a normal distribution with mean `f` and variance σ.
+"""
+struct HeteroscedasticGaussianLikelihood end
+
+@functor HeteroscedasticGaussianLikelihood
```

Comment:
No need for this since there are no parameters:
```suggestion

@functor HeteroscedasticGaussianLikelihood
```